lib/rules/vars-on-top.js

```diff
@@ -28,6 +28,15 @@ module.exports = function (context) {
     }
 
     /**
+     * Check to see if its a ES6 import declaration
+     * @param {ASTNode} node - any node
+     * @returns {Boolean} whether the given node represents a import declaration
+     */
+    function looksLikeImport(node) {
+        return node.type === "ImportDeclaration";
```

Comment:
ESTree split this up into multiple node types - see https://github.com/estree/estree/blob/master/es6.md#modules, though I don't recall whether Espree implements them yet. Should be able to check them all here for forward-compatibility even if they aren't parsed that way yet.